server/controllers/popController.ts

```diff
@@ -0,0 +1,141 @@
+      next(error)
+    }
+  }
+}
+
+export const renderPopProgress = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const progressData = await popService.getProgressDetails()
+      res.render('pages/pop/progress', progressData)
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const renderMessageThread = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const message = await popService.getMessageById(req.params.id)
+      if (message) {
+        res.render('pages/pop/messageThread', { message, error: req.flash('error') })
+      } else {
+        res.status(404).send('Message not found')
+      }
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const handleMessageThread = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const messageId = req.params.id
+      const messageText = req.body.message
+      const fileData = req.file
+      const userId = req.session.user_id
+
+      const success = await popService.addMessageToThread(messageId, messageText, fileData, userId)
+      if (success) {
+        res.redirect(`/pop/messages/thread/${messageId}`)
+      } else {
+        res.status(404).send('Message not found')
+      }
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const renderAppointments = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const appointments = await popService.getAppointments()
+      res.render('pages/pop/appointments', appointments)
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const renderConditions: RequestHandler = async (req, res, next) => {
+  try {
+    res.render('pages/pop/conditions')
+  } catch (error) {
+    next(error)
+  }
+}
+
+export const renderViewAppointment: RequestHandler = async (req, res, next) => {
+  try {
+    res.render('pages/pop/view-appointment')
+  } catch (error) {
+    next(error)
+  }
+}
+
+export const renderViewPastAppointment: RequestHandler = async (req, res, next) => {
+  try {
+    res.render('pages/pop/view-past-appointment')
+  } catch (error) {
+    next(error)
+  }
+}
+
+export const renderMessages = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const messages = await popService.getAllMessages()
+      res.render('pages/pop/messages', { messages })
+    } catch (error) {
+      next(error)
+    }
+  }
+}
+
+export const handleNewMessage = (popService = getPopService()): RequestHandler => {
+  return async (req, res, next) => {
+    try {
+      const { subject, message: messageText } = req.body
+      const fileData = req.file
+      const userId = req.session.user_id
+
+      if (subject && messageText) {
+        const messageId = await popService.createNewMessage(subject, messageText, fileData, userId)
```

Comment:
We might want to fake the concept of recipient in this call - POP might be able to only send to one person but the system need to know who that is somehow, somewhere. Most important for new messages